### ISSUE TITLE
Model takes several minutes to download occasionally

### ISSUE THREAD

**Issue by espressotechie** [fofr/cog-comfyui#154]
When running my json workflow through api in my application, most of the times it downloads the required models and runs within a minute. However at times it takes several minutes and when I checked the logs, it showed that it took 4 minutes to download a model. Why does this happen occasionally and is there a way to ensure that if the models take more than a minute to download, the prediction is cancelled or errored out? This extra time adds to my billing :(
![scrnsht01](https://github.com/user-attachments/assets/af3fe86e-7a5f-457a-9eb0-91080165f9bf)

**Comment by aichromator** [comment]
Yes, same issue. especially when multiple models are options available for a custom node, it loads all. How do we solve it?

**Comment by fofr** [comment]
My best guess here is that this is a cache miss on Replicate's side, meaning the weights download slower. Any subsequent requests shouldn't have that problem.

**Comment by bertagknowles** [comment]
Hmm, then I think I'll create an automatic cancel after a set time in the application side.

**Comment by bertagknowles** [comment]
The problem is that if I use a timer to cancel requests after one minute, then the requests that are waiting in queue will also be cancelled since sometimes in public models the queue can also take more time. Will it be possible for you to add in the backend that if the download time exceeds one minute or if the cache is missed then the request should time out/fail?

**Comment by bertagknowles** [comment]
Got this error and was billed unnecessarily....something is broken....this is not reliable for api

![error](https://github.com/user-attachments/assets/ac732dfd-e947-425c-90b8-b2fb42dff43a)

**Comment by fofr** [comment]
@bertagknowles FYI, you are not billed for failed predictions. You won't be billed for this.

**Comment by bertagknowles** [comment]
Oh, that's a big relief...thanks for letting me know.

**Comment by bertagknowles** [comment]
However, there is something wrong with the model lately and the output file is not being shown (blank). 
![error](https://github.com/user-attachments/assets/c4b118c7-71c0-46ac-bd84-67b56ac27f95)

**Comment by fofr** [comment]
Closing as I think this is now resolved.
